packages/ra-ui-materialui/src/input/ReferenceArrayInput.tsx

```diff
@@ -120,7 +122,12 @@ ReferenceArrayInput.propTypes = {
     source: PropTypes.string,
 };
 
-const defaultChildren = <AutocompleteArrayInput />;
+const defaultChildren = (readOnly, disabled) => (
+    <AutocompleteArrayInput
+        readOnly={readOnly}
+        disabled={disabled || readOnly}
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
**issue:** ‏You should only pass `disabled` to `AutocompleteArrayInput` if the `disabled` prop is truthy, otherwise when the user sets only `readOnly`, the value will still be absent from the form values, which defeats the purpose of this PR. TLDR: let `AutocompleteArrayInput`  decide what to do with the props. Just forward them as-is.

```suggestion
        disabled={disabled}
```